plugin/trino-redis/src/main/java/io/trino/plugin/redis/RedisRecordCursor.java

```diff
@@ -306,7 +305,7 @@ private boolean fetchKeys()
                 }
                 break;
                 case ZSET:
-                    Set<String> keys = jedis.zrange(split.getKeyName(), split.getStart(), split.getEnd());
+                    List<String> keys = jedis.zrange(split.getKeyName(), split.getStart(), split.getEnd());
```

Comment:
very misleading decision from Jedis. It does return a set since sorted-sets in redis can't contain duplicate values.

Reply:
I'm not really sure here. As far as I can see, they deliberately changed  APIs to return List<String> and it's consistent across the client

Reply:
I'll recheck then. IIRC sorted sets have a key which hold a pair of value + score. the values must be unique.

Reply:
Ah, I see their justification. There are some methods on sorted-sets which can return duplicate values (even if it can't store duplicates). They changed for consistency across the API (and preserving order of returned values which is important for a sorted set).

In this case `ZRANGE` can never return duplicates so I'd still like to wrap this using the Guava ImmutableSet builder (to catch if we ever get duplicates here - instead of silently swallowing them). However since the only thing we are doing here is returning an iterator I'm fine with existing code (maybe remove the variable entirely?).